scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook

```diff
@@ -0,0 +1,456 @@
+        printf "[ERROR] ${RED}secret-gen controller is not installed in the secretgen-controller namespace (this is if you are using Tanzu RabbitMQ commercial version only) ${END}\\n"
+        exit -1
+    fi
+
+    echo "kapp version:                              $kapp_version" > $OUTPUT_DIRECTORY/carvel/info/carvel_versions.info
+    echo "secretgen-controller version:              $secret_gen_controller_version" >> $OUTPUT_DIRECTORY/carvel/info/carvel_versions.info
+
+    printf "\\n[INFO] kapp-controller and secret-gen controller versions saved in $OUTPUT_DIRECTORY/carvel/info/carvel_versions.log\\n"
+}
+
+data_diagnostic_carvel_status()    {
+
+    set -e
+
+    # kapp status namespace
+    printf "\\n[INFO] Looking for kapp objects status and definitions in kapp-controller namespace and saving it to a file.\\n"
+    kubectl get all -n $KAPP_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/kapp_get_all_objects_status.info
+    kubectl describe replicaset -n $KAPP_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/kapp_get_all_objects_status.info
+    kubectl describe deploy -n $KAPP_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/kapp_get_all_objects_status.info
+    kubectl get deploy -n $KAPP_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/kapp-controller-definition.yaml
+
+
+    # secretgen-controller status namespace
+    printf "[INFO] Looking for secretgen-controller objects status and definitions in secretgen-controller namespace and saving it to files in $OUTPUT_DIRECTORY/carvel/info.\\n"
+    kubectl get all -n $SECRETGEN_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl describe replicaset -n $SECRETGEN_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl describe deploy -n $SECRETGEN_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl get deploy -n $SECRETGEN_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/secretgen-controller-definition.yaml
+
+    # secretgen-controller status namespace
+    printf "[INFO] Looking for PackageRepo and PackageInstall objects status in secretgen-controller namespace and saving it to files in $OUTPUT_DIRECTORY/carvel/info.\\n"
+    kubectl get PackageRepository -n $CARVEL_PACKAGES_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/packagerepo.info
+    kubectl describe PackageRepository -n $CARVEL_PACKAGES_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/packagerepo.info
+    kubectl get PackageRepository -n $CARVEL_PACKAGES_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/packagerepo_definition.yaml
+    kubectl get PackageInstall -n $CARVEL_PACKAGES_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/packageinstall.info
+    kubectl describe PackageInstall -n $CARVEL_PACKAGES_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/packageinstall.info
+    kubectl get PackageInstall -n $CARVEL_PACKAGES_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/packageinstall_definition.yaml
```

Comment:
I think these should fail if they don't exist - if we've specifically asked for carvel information and we don't get any then that should be raised as an error. In my tests I didn't notice I hadn't specified the right namespace.

Reply:
Done! the script will now provide an ERROR log if PackageRepository or PackageInstall are not present in the provided namespace!